docs/upgrading_binary.rst

```diff
@@ -0,0 +1,190 @@
+Upgrading Binary(Set)Attribute
+==============================
+
+.. warning::
+
+    The behavior of :py:class:`~pynamodb.attributes.BinaryAttribute` and
+    :py:class:`~pynamodb.attributes.BinarySetAttribute` has changed in backwards-incompatible ways
+    as of the 6.0 release of PynamoDB.
+
+    To prevent data corruption, use :code:`legacy_encoding=True` for **existing** binary attributes.
+
+Context
+#######
+
+PynamoDB version 5 (and lower) had two bugs in the way they handled binary attributes,
+which were addressed in PynamoDB 6:
+
+- Top-level binary attributes (i.e. within a :py:class:`~pynamodb.models.Model`) were being Base64-encoded
+  twice. For elements in BinarySetAttribute, each element was being encoded twice.
+
+  This behavior was an oversight and resulted in larger item sizes and non-standard semantics.
+
+  .. admonition:: For example...
+
+     The 4 bytes :code:`CA FE F0 0D` should've been sent over the wire as :code:`yv7wDQ==` (single round
+     of Base64 encoding). Server-side, they would've been decoded and stored as 4 bytes. Instead, they were put through an extra
+     round of Base64 encoding, thus sending :code:`eXY3d0RRPT0=` over the wire. Server-side, they decoded into :code:`yv7wDQ==`
+     (in bytes, :code:`79 76 37 77 44 51 3D 3D`) and stored as 8 bytes.
+
+- Nested binary attributes (i.e. within a :py:class:`~pynamodb.attributes.MapAttribute` and :py:class:`~pynamodb.attributes.ListAttribute`)
+  were being wrapped in an additional layer of Base64 encoding on every serialization roundtrip.
+
+  Not only it prevented them from being deserialized correctly, but also the model would also grow
+  in size exponentially until it hit the DynamoDB item limit of 400KB. For this reason we conclude
+  that :code:`BinaryAttribute` and :code:`BinarySetAttribute` were not used in practice within maps and lists
+  before PynamoDB 6.0 and thus will not require any legacy encoding.
```

Comment:
> not used in practice within maps and lists... and thus will not require any legacy encoding

Do we need to differentiate between raw and typed maps here? Given that the example below does mention specifying `legacy_encoding` on a map attr.

Reply:
Good point. I'll use your phrasing here.